### DISCUSSION TITLE
Instance Segmentation support in AugmentationSequential

### DISCUSSION THREAD

**Discussion by ashnair1** [kornia/kornia#2417] [Questions]
The following is the default example for using `AugmentationSequential` with boxes and masks 

```python
import torch
import kornia
from kornia.augmentation import AugmentationSequential
from kornia.geometry.boxes import Boxes

input = torch.randn(2, 3, 256, 256)
mask = torch.ones(2, 3, 256, 256)

bbox = torch.tensor([[
    [1., 1.],
    [2., 1.],
    [2., 2.],
    [1., 2.],
]]).expand(2, 1, -1, -1)

aug_list = AugmentationSequential(
    kornia.augmentation.ColorJiggle(0.1, 0.1, 0.1, 0.1, p=1.0),
    kornia.augmentation.RandomHorizontalFlip(p=1.0),
    data_keys=["input", "mask", "bbox"],
    same_on_batch=False,
    random_apply=10,
)
out = aug_list(input, mask, bbox)
```

In instance segmentation, we'll be working with **list of boxes and masks** as each image can have N detections. So the example would now be as follows:

```python
import torch
import kornia
from kornia.augmentation import AugmentationSequential
from kornia.geometry.boxes import Boxes

input = torch.randn(2, 3, 256, 256)
mask = [
    torch.ones(3, 256, 256),
    torch.ones(2, 256, 256)
]

bbox = [
    torch.tensor([[ 28.,  53., 143., 164.],
                  [254., 158., 364., 290.],
                  [307., 204., 413., 350.]]),
    torch.tensor([[254., 158., 364., 290.],
                  [307., 204., 413., 350.]]),
]

bbox = [Boxes.from_tensor(i).data for i in bbox]


aug_list = AugmentationSequential(
    kornia.augmentation.ColorJiggle(0.1, 0.1, 0.1, 0.1, p=1.0),
    kornia.augmentation.RandomHorizontalFlip(p=1.0),
    data_keys=["input", "mask", "bbox"],
    same_on_batch=False,
    random_apply=10,
)
out = aug_list(input, mask, bbox)
```

Running this script, it can be seen that while list of boxes works with AugmentationSequential, list of masks do not. Is there a way within Kornia to handle list of masks?
